test/factories/units_factory.rb

```diff
@@ -7,7 +7,16 @@
     meeting_time      "17:30"
     meeting_location  "ATC101"
     sequence(:abbreviation) { |n| "LA1-#{n}" }
+    unit_activity_set
+  end
+
+  factory :unit_activity_set do
     unit
+    activity_type
+
+    after(:create) do | unit_activity_set |
+      create_list(:tutorial, 3, unit_activity_set: unit_activity_set)
```

Comment:
it would be good to have a way to control what the generated activity set looks like.

Reply:
This can be done through the associated data - so not sure this is needed here. We just need some default that can be overridden.